firesim/firesim
# [WIP] Parameterized NIC

Things added
- Added support for a parameterized NIC (64b, 128b, 256b).
- Updated documentation to reflect the changes and how to apply them
- Support in config.ini's and TargetConfigs to change it

Tested with:
- run-bw-test.sh
- run-simperf-test-latency.sh
- run-ping-latency.sh
- example_2config with one instance sending packets to the other

Questions:
- I removed the const of TOKEN_QUEUE_DEPTH in favor of individual queue depths for both TracerV and the NIC. Is this fine or should these be linked/the same?
- Should the benchmark .py's be changed to accommodate the different flit sizes (for ex. `deploy/workloads/bw-test-two-instances/bw-test-graph.py` is "hardcoded" to use 64b flits)
